src/index.ts

```diff
@@ -63,11 +69,80 @@ function successLoader(
       : rawFilePath;
 
   const fileVersion = updateFileInCache(filePath, contents, instance);
+  const referencedProject = getProjectReferenceForFile(filePath, instance);
+  if (referencedProject) {
+    if (referencedProject.commandLine.options.outFile) {
+      throw new Error(
+        `The referenced project at ${
+          referencedProject.sourceFile.fileName
+        } is using the ` +
+          `outFile' option, which is not supported with ts-loader.`
+      );
+    }
 
-  const { outputText, sourceMapText } = options.transpileOnly
-    ? getTranspilationEmit(filePath, contents, instance, loader)
-    : getEmit(rawFilePath, filePath, instance, loader);
+    const jsFileName = instance.compiler.getOutputJavaScriptFileName(
```

Comment:
given that the whole tsbuild API is now internal, this uses an internal function? I couldn't find a custom implementation if it in this PR